Add description for UpgradeLog enhancement (#349)
harvester/docs: docs/upgrade/troubleshooting.md

    ```

4. Click the upgrade button in the Harvester dashboard to start an upgrade again.

### Download upgrade relevant logs

> We also designed and implemented a mechanism for collecting all the upgrade-related logs along with the upgrade procedure automatically. By default, this is enabled, users can choose to opt out of such behavior though.

Reviewer: ```suggestion
We have designed and implemented a mechanism to automatically collect all the upgrade-related logs and display the upgrade procedure. By default, this feature is enabled. You can also choose to opt out of such behavior.
```

![The "Enable Logging" checkbox on the upgrade confirmation dialog](/img/v1.2/upgrade/enable_logging.png)

> Users can download the log archive during an upgrade.

Reviewer: ```suggestion
You can click the **Download Log** button to download the log archive during an upgrade.
```

![Download the upgrade log archive by clicking the "Download Log" button on the upgrade dialog](/img/v1.2/upgrade/download_upgradelog_dialog.png)

> Log entries will be collected as files for each upgrade-related Pod, even for intermediate Pods. The support bundle provides a snapshot of the current state of the cluster, including logs and resource manifests, while the upgrade log preserves any logs generated during an upgrade. By combining these two, we can look further into issues during upgrades.

Reviewer: ```suggestion
Log entries will be collected as files for each upgrade-related Pod, even for intermediate Pods. The support bundle provides a snapshot of the current state of the cluster, including logs and resource manifests, while the upgrade log preserves any logs generated during an upgrade. By combining these two, you can further investigate the issues during upgrades.
```

![The upgrade log archive contains all the logs generated by the upgrade-related Pods](/img/v1.2/upgrade/upgradelog_archive.png)

> After the upgrade ended, Harvester stops collecting the upgrade log automatically so it will not continue to grow and occupy the disk space. It's also worth noting that the upgrade log will be purged once users click on the "Dismiss it" button on the upgrade dialog.

Reviewer: ```suggestion
After the upgrade ended, Harvester stops collecting the upgrade logs to avoid occupying the disk space. In addition, you can click the **Dismiss it** button to purge the upgrade logs.
```

After the upgrade ended, Harvester stops collecting the upgrade logs to avoid occupying the disk space. In addition, you can click the **Dismiss it** button to purge the upgrade logs.

![The upgrade log archive contains all the logs generated by the upgrade-related Pods](/img/v1.2/upgrade/dismiss_upgrade_to_remove_upgradelog.png)

For more details, please refer to the [upgrade log HEP](https://github.com/harvester/harvester/blob/master/enhancements/20221207-upgrade-observability.md).

:::caution

The storage volume for storing upgrade-related logs is 1GB by default. If an upgrade went into issues, the logs may consume all the available space of the volume. To work around such kind of incidents, try the following steps:

1. Detach the `log-archive` Volume by scaling down the `fluentd` StatefulSet and `downloader` Deployment.

```
# Locate the StatefulSet and Deployment
$ kubectl -n harvester-system get statefulsets -l harvesterhci.io/upgradeLogComponent=aggregator
NAME                                               READY   AGE
hvst-upgrade-xxxxx-upgradelog-infra-fluentd   1/1     43s

> $ kubectl -n harvester-system get deployments -l harvesterhci.io/upgradeLogComponent=downloader

Reviewer 2: It would be nice to add an empty line between the next command and the result (much clear to read).

NAME                                            READY   UP-TO-DATE   AVAILABLE   AGE
hvst-upgrade-xxxxx-upgradelog-downloader   1/1     1            1           38s

# Scale down the resources to terminate any Pods using the volume
$ kubectl -n harvester-system scale statefulset hvst-upgrade-xxxxx-upgradelog-infra-fluentd --replicas=0
statefulset.apps/hvst-upgrade-xxxxx-upgradelog-infra-fluentd scaled
$ kubectl -n harvester-system scale deployment hvst-upgrade-xxxxx-upgradelog-downloader --replicas=0
deployment.apps/hvst-upgrade-xxxxx-upgradelog-downloader scaled
```

> 2. Expand the volume size via Longhorn dashboard. For more details, please refer to [the volume expansion guide](https://longhorn.io/docs/1.3.2/volumes-and-nodes/expansion/).

Reviewer 2: Please add some tips on how to find the volume.
